pkg/kafka/publisher.go

```diff
@@ -11,35 +11,66 @@ import (
 )
 
 type Publisher struct {
-	producer  sarama.SyncProducer
-	marshaler Marshaler
-
-	logger watermill.LoggerAdapter
+	config   PublisherConfig
+	producer sarama.SyncProducer
+	logger   watermill.LoggerAdapter
 
 	closed bool
 }
 
 // NewPublisher creates a new Kafka Publisher.
 func NewPublisher(
-	brokers []string,
-	marshaler Marshaler,
-	overwriteSaramaConfig *sarama.Config,
+	config PublisherConfig,
 	logger watermill.LoggerAdapter,
 ) (message.Publisher, error) {
-	if overwriteSaramaConfig == nil {
-		overwriteSaramaConfig = DefaultSaramaSyncPublisherConfig()
+	config.setDefaults()
+
+	if err := config.Validate(); err != nil {
+		return nil, err
 	}
 
 	if logger == nil {
 		logger = watermill.NopLogger{}
 	}
 
-	producer, err := sarama.NewSyncProducer(brokers, overwriteSaramaConfig)
+	producer, err := sarama.NewSyncProducer(config.Brokers, config.OverwriteSaramaConfig)
 	if err != nil {
 		return nil, errors.Wrap(err, "cannot create Kafka producer")
 	}
 
-	return &Publisher{producer, marshaler, logger, false}, nil
+	return &Publisher{
+		config:   config,
+		producer: producer,
+		logger:   logger,
+	}, nil
+}
+
+type PublisherConfig struct {
+	// Kafka brokers list.
+	Brokers []string
+
+	// Marshaler is used to marshal messages from Watermill format into Kafka format.
+	Marshaler Marshaler
+
+	// OverwriteSaramaConfig holds additional sarama settings.
+	OverwriteSaramaConfig *sarama.Config
```

Comment:
In other pubsubs, we don't call the 3rd party configs OverwriteX, just X.
So this may be a bit confusing (this was the first config that we did and there was nothing to compare it to).

Reply:
True, I think we can go with `SaramaConfig`. @roblaszczak ?